pyvisgraph/visible_vertices.py

```diff
@@ -287,7 +288,8 @@ def angle2(point_a, point_b, point_c):
     a = (point_c.x - point_b.x)**2 + (point_c.y - point_b.y)**2
     b = (point_c.x - point_a.x)**2 + (point_c.y - point_a.y)**2
     c = (point_b.x - point_a.x)**2 + (point_b.y - point_a.y)**2
-    return acos((a + c - b) / (2 * sqrt(a) * sqrt(c)))
+    cos_value = (a + c - b) / (2 * sqrt(a) * sqrt(c))
+    return acos( max(-1, min(1, cos_value)) )

```

Comment:
In functions that are called a lot, I like to avoid using functions like `max`, `min`, `round`. Its slow and when you have functions that are run millions of times, its best to keep it all math if you can. The problem here I think is a rounding error, take a look at 9c0ec2212f513 for a way to solve rounding without using math functions.

Reply:
Well, since it's just a numerical issue for v ~= 1 or v ~= -1, would you prefer this one ?
```python
v = (a + c - b) / (2 * sqrt(a) * sqrt(c))
return acos( 1 if v > 1 else -1 if v < -1 else v )
```

Reply:
Not sure about the optimal optimization to use, but the int cast may not be applicable in this case.

Reply:
I did some performance tests, using a world map containing 3,846 edges, which results in 88,887 edges in the visibility graph. 
* Without changing `angle2`, it builds the graph in 547.21 seconds. 
* Using `min` and `max`, it runs in 726.70 seconds (a 32.8% performance hit).
* Using `if` statements, it runs in 568.76 seconds (a 3.9% performance hit).
* Using the `int` cast truncating solution, it runs in 548.80 seconds (negligible change to performance).

https://github.com/TaipanRex/pyvisgraph/issues/19#issuecomment-312628047 explains floating point representation errors, which this is actually a case of. 

e9eeeaa I thought I could get away with it, but you can see I mention that tolerance may be needed in angle2 as well.

So I suggest using, for best performance:
```python
cos_value = (a + c - b) / (2 * sqrt(a) * sqrt(c))
return acos(int(cos_value*T)/T2)
```

Reply:
thats because `COLIN_TOLERANCE` is set to 13 on line 31 of visible_vertices.py, if you change that to 12, you wont run into the problem.

Reply:
Hugh... But how safe is that ?
with `COLIN_TOLERANCE` set to 12 I still found -1.0000000000075326 casted to -1.000000000007

Reply:
What about an exception ?
```python
cos_value = (a + c - b) / (2 * sqrt(a) * sqrt(c))
try:
    return acos(cos_value)
except ValueError:
    return 0 if cos_value > 1 else pi
```

Reply:
Its a balancing act between safety and performance I guess. For a layman user it might be too complicated to figure out the right tolerance for the coordinate system you are using. I set the default value too high (13), I doubt you would need 13 digits of precision. By setting it to something like 8, that would cover even extreme cases. 

From my performance numbers, we are talking  almost 10 minutes to build the visibility graph. Every percent performance improvement counts.

Reply:
And I guess an exception handling would also impact performances too much, right?

So, can I add an optional parameter in VisGraph constructor to tune `COLIN_TOLERANCE`?

Reply:
@deboc I merged 757a4342d5b3dca17c1356c12313d35bef809ae3 for polygon_crossing() and  2dd5bbea7d83c67a33ed7eecc40902cc66379f2c for angle2(). I plan on implementing a config solution, see #29.